File: nomad-samples/js/contents-api/contents/update-content.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function updateContent(AUTH_TOKEN, CONTENT_DEFINITION_ID, ID, PROPERTIES) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Build the payload body
    const BODY = {
        contentDefinitionId: CONTENT_DEFINITION_ID,
        contentId: ID,
        properties: PROPERTIES
    };

    // Send POST request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/content/${ID}`, {
        method: "PUT",
        headers: HEADERS,
        body: JSON.stringify(BODY),
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the response
        const ID = await RESPONSE.text();

        // Return the ID
        return ID.replaceAll('"', "");
    }

    apiExceptionHandler(RESPONSE, "Create/Update movie failed")
}

File: nomad-samples/admin/py/content_api/contents/update_content.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def update_content(AUTH_TOKEN, ID, CONTENT_DEFINITION_ID, PROPERTIES):
    # Create header for the request
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    API_URL = f"{ADMIN_API_URL}/content/{ID}"
    
    HEADERS = {
      "Content-Type": "application/json",
      "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload body
    BODY = {
        "contentDefinitionId": CONTENT_DEFINITION_ID,
        "contentId": ID,
        "properties": PROPERTIES
    }

    try:
        # Send POST request
        RESPONSE = requests.put(API_URL, headers=HEADERS, data=json.dumps(BODY))

        # Check for success
        if (not RESPONSE.ok):
            raise Exception()
        
        # Get the response
        ID = RESPONSE.json()

        # Return the ID
        return ID
    
    except:
        api_exception_handler(RESPONSE, "Create/Update movie failed")

